File: java/src/main/java/de/xjustiz/core/models/Dokumentklasse.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlAttribute;
import jakarta.xml.bind.annotation.XmlElement;
import org.jetbrains.annotations.Nullable;

/**
 * Die Dokumentklasse definiert den Typ eines Dokuments (z. B. Klageschrift, Urteil).
 * <p>
 *  <u><b>Document class:</b></u> The document class defines the type of a document (e.g., statement of claim, judgment).
 */
@XmlAccessorType(XmlAccessType.FIELD)
public class Dokumentklasse {

    /**
     * Die Dokumentklasse definiert den Typ eines Dokuments (z. B. Klageschrift, Urteil).
     * <p>
     *  <u><b>Document class:</b></u> The document class defines the type of a document (e.g., statement of claim, judgment).
     */
    @XmlAttribute(name = "listVersionID")
    @JacksonXmlProperty(isAttribute = true, localName = "listVersionID")
    @JsonProperty("ListVersionId")
    private String listVersionId;

    /**
     * Die Dokumentklasse definiert den Typ eines Dokuments (z. B. Klageschrift, Urteil).
     * <p>
     *  <u><b>Document class:</b></u> The document class defines the type of a document (e.g., statement of claim, judgment).
     */
    @XmlAttribute(name = "listURI")
    @JacksonXmlProperty(isAttribute = true, localName = "listURI")
    @JsonProperty("ListUri")
    @Nullable
    private String listUri;

    /**
     * Hilfseigenschaft für die XML-Serialisierung.
     * <p>
     *  <u><b>Code for XML:</b></u> Helper property for XML serialization.
     */
    @XmlElement(name = "code", namespace = "http://www.xjustiz.de")
    @JacksonXmlProperty(localName = "code", namespace = "http://www.xjustiz.de")
    @JsonProperty("CodeForXml")
    @Nullable
    private String codeForXml;

    /**
     * Default constructor.
     */
    public Dokumentklasse() {
    }

    /**
     * Die Dokumentklasse definiert den Typ eines Dokuments (z. B. Klageschrift, Urteil).
     *
     * @return the listVersionId
     */
    public String getListVersionId() {
        return listVersionId;
    }

    /**
     * Sets the listVersionId.
     *
     * @param listVersionId the listVersionId to set
     */
    public void setListVersionId(String listVersionId) {
        this.listVersionId = listVersionId;
    }

    /**
     * Die Dokumentklasse definiert den Typ eines Dokuments (z. B. Klageschrift, Urteil).
     *
     * @return the listUri
     */
    @Nullable
    public String getListUri() {
        return listUri;
    }

    /**
     * Sets the listUri.
     *
     * @param listUri the listUri to set
     */
    public void setListUri(@Nullable String listUri) {
        this.listUri = listUri;
    }

    /**
     * Hilfseigenschaft für die XML-Serialisierung.
     *
     * @return the codeForXml
     */
    @Nullable
    public String getCodeForXml() {
        return codeForXml;
    }

    /**
     * Sets the codeForXml.
     *
     * @param codeForXml the codeForXml to set
     */
    public void setCodeForXml(@Nullable String codeForXml) {
        this.codeForXml = codeForXml;
    }

}

File: sdk/dotnet/src/xjustiz.core-dotnet/Models/Entities/Dokumentklasse.cs
namespace xjustiz.core_dotnet.Models.Entities;

using System.Text.Json.Serialization;
using System.Xml.Serialization;
using xjustiz.core_dotnet.Models.Codes;
using xjustiz.core_dotnet.Models.Codes.Dokumentklasse;
using xjustiz.core_dotnet.Models.Helpers;
using xjustiz.core_dotnet.Util.Versioning;

/// <summary>
/// Die Dokumentklasse definiert den Typ eines Dokuments (z. B. Klageschrift, Urteil).<br/>
/// <u><b>Document class:</b></u> The document class defines the type of a document (e.g., statement of claim, judgment).
/// </summary>
[XJustizAvailability(XJustizVersion.V2_2_1)]
[XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
public class Dokumentklasse : ICode<DokumentklasseCode>
{
    /// <inheritdoc/>
    [XmlAttribute("listVersionID")]
    public string ListVersionId { get; set; } = DokumentklasseCodeLists.LatestList.Version;

    /// <inheritdoc/>
    [XmlAttribute("listURI")]
    public string? ListUri { get; set; } = DokumentklasseCodeLists.Uri;

    /// <summary>
    /// Ruft den Dokumentklasse-Code ab oder legt diesen fest.<br/>
    /// <u><b>Code:</b></u> Gets or sets the document class code.
    /// </summary>
    [XmlIgnore]
    [JsonPropertyName("code")]
    public DokumentklasseCode Code { get; set; }

    /// <summary>
    /// Hilfseigenschaft für die XML-Serialisierung.<br/>
    /// <u><b>Code for XML:</b></u> Helper property for XML serialization.
    /// </summary>
    [XmlElement("code", Namespace = SchemeConstants.XJustizCore_Tns)]
    [JsonIgnore]
    public string CodeForXml
    {
        get => Code.ToCode();
        set => Code = DokumentklasseCodeMapper.TryParse(value, out var result) ? result : default;
    }
}
